### DISCUSSION TITLE
플러그인 상호 의존성이란 무엇일까?

### DISCUSSION THREAD

**Discussion by CODe5753** [Well-Founded-Confidence/RealMySQL#16] [Q&A] [answered]
> "플러그인은 상호 의존 관계를 설정할 수 없어서 초기화가 어렵다"

플러그인 설명을 보면서 "플러그인 상호 의존성"이라는 말이 나오더라구요.

플러그인 내용 자체를 처음 보는거라 좀 이해가 안됐는데,
이 내용이 무슨 말일까요?

**Comment by nahyunLee** [comment, marked as answer]
저도 같은 내용이 한번에 이해가지 않아서 고민해보았는데요, 아래와 같은 예시로 이해했습니당🤔

-  A와 B가 MySQL서버를 통해서 간접적으로 상호관계를 맺고 있지만 직접적으로 연결(설정)이 되어있지 않음.
- 상호관계를 맺고 있기 때문에 A를 초기화 하면 B까지 영향을 받음
-  하지만 직접적으로 연결이 안되어있으니, A가 초기화 되더라도 B가 직접적으로 'A가 초기화 되었다는 사실'을 알기가 힘듦
-  B가 'A가 초기화 되었다는 사실'을 알 수 없어 대응이 힘들지만, 결국 변화에 영향을 받으니 A를 함부로 초기화하기 어려움

**Reply by CODe5753** [reply]
네 저도 나현님 말씀하신대로 이해했습니다!!
내용 공유 감사드려요 🙌
